include/ua_plugin_securitypolicy.h

```diff
@@ -178,32 +178,34 @@ typedef struct
     const size_t encryptingBlockSize;
 } UA_SecurityPolicySymmetricModule;
 
+/**
+ * This struct defines initialization Data that is potentially needed by every security policy.
+ * Some of these, like the revocation list for example, may be omitted (empty ByteString).
+ */
+typedef struct
+{
+    UA_ByteString localPrivateKey;
+    UA_ByteString localCertificate;
+    UA_ByteString certificateTrustList;
+    UA_ByteString certificateRevocationList;
+} UA_Endpoint_SecurityContext_RequiredInitData;
+
 struct UA_Endpoint_SecurityContext {
+    /**
+     * \brief Creates a new endpoint context with the supplied initialization data.
+     *
+     * \param securityPolicy the securityPolicy this function is invoked on.
+     * \param initData the required initialization data for the endpoint context.
+     */
     UA_StatusCode (*const newContext)(const UA_SecurityPolicy *securityPolicy,

```

Comment:
Hmm, so you mean we have multiple policy endpoints, one for each local certificate/private key and policy, and endpoints can use these (potentially multiple endpoints the same policyContext).
That might be an option i guess.


I'm still not sure if this is all even possible, since i couldn't find anything about that in the standard, but it might be useful to have in the future.